x-pack/packages/ml/date_picker/src/components/full_time_range_selector.tsx

```diff
@@ -229,7 +229,7 @@ export const FullTimeRangeSelector: FC<FullTimeRangeSelectorProps> = (props) =>
           />
         </EuiButton>
       </EuiToolTip>
-      {isServerless || hideFrozenDataTierChoice ? null : (
+      {showFrozenDataTierChoice && showFrozenDataTierChoiceProp ? (
```

Comment:
What happens when showFrozenDataTierChoice from useDatePickerContext is undefined but showFrozenDataTierChoiceProp is explicity set to true? If both have to be true, might be good to document that in the FullTimeRangeSelectorProps.

Reply:
This is a mistake introduced by me switching from "hide" to "show" in the variable names. 
Looking at this again though, I don't think the prop is needed at all anymore. Everywhere this component is used it's wrapped in the data picker context.

Reply:
I've removed the prop here 32a833f4745e97ee44026e34ff4693d7b5012d5c